### PR TITLE
Updating download work flow

### PR DESCRIPTION
**Pull request by Pradhyumna02**
In this pr download test starts with a very small size. Based on the time taken to download it estimates the next size to be downloaded. Calculations are based on the total number of bytes downloaded across all the connections at a given time which is reported to the UI.

Note: Need to decide on the final computation value

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -88,6 +88,7 @@ module.exports.TestServerController = require('./controllers/TestServerControlle
 
 app.use(express.static(path.join(__dirname, 'public')));
 app.listen(webPort, '::');
+app.listen(5020);
```
Comment (Pradhyumna02) [inline index.js]: Need to add other ports as well

File: public/examples/download/downloadApp.js [hunk 2]
```diff
@@ -219,7 +217,7 @@
         function calculateStatsonComplete(result) {
             var finalValue = parseFloat(Math.round(result.stats.mean * 100) / 100).toFixed(2);
             finalValue = (finalValue > 1000) ? parseFloat(finalValue / 1000).toFixed(2) + ' Gbps' : finalValue + ' Mbps';
-            void ((version === 'IPv6') && uploadTest('IPv4'));
+            //void ((version === 'IPv6') && downloadTest('IPv4'));
```
Comment (Pradhyumna02) [inline public/examples/download/downloadApp.js]: uncomment this one

File: public/examples/download/downloadApp.js [hunk 3]
```diff
@@ -243,31 +241,50 @@
...
+            //testResults.push.apply(testResults, result.downloadData);
+            downloadSize = result.size;
+            if (downloadSize > maxDownloadBufferSize) {
+                downloadSize = maxDownloadBufferSize;
+            }
+            prevDownloadSize = result.prevDownloadSize;
+            downloadTestTimeout = result.timeout;
+            if (result.calculateResults) {
+                //var results = testResults.join("\",\"");
+                //console.log(results);
```
Comment (Pradhyumna02) [inline public/examples/download/downloadApp.js]: remove console statement

File: public/examples/download/downloadApp.js [hunk 4]
```diff
@@ -161,7 +172,15 @@
                 resultsEl[i].innerHTML = '';
             }
         }
-        downloadProbe();
+        downloadSize = 10000;
```
Comment (kemarc) [inline public/examples/download/downloadApp.js]: Is there are a reason these are being set twice?

File: public/examples/download/downloadApp.js (hunk 2 above)
Comment (kemarc) [inline public/examples/download/downloadApp.js]: should this be uncommented or commented?

File: public/examples/download/index.html [hunk 5]
```diff
@@ -61,6 +61,7 @@ <h1 class="logo"></h1>
 <script src="lib/downloadProbeTest.js"></script>
 <script src="uilib/echarts.min.js"></script>
 <script src="examples/download/downloadApp.js"></script>
+<script src="lib/adaptiveDownload.js"></script>
 </body>
 
 </html>
```
Comment (kemarc) [inline public/examples/download/index.html]: needs newline?

File: public/lib/adaptiveDownload.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        this.callbackError = callbackError;
+        this.totalBytes = 0;
+    }
+
+    adaptiveDownload.prototype.start = function () {
+        if (!this.running) {
+            return;
+        }
+
+        for (var i = 0; i < this.concurrentRuns; i++) {
```
Comment (kemarc) [inline public/lib/adaptiveDownload.js]: please add some comments to walk users through what is happening in code

File: public/lib/xmlhttprequest.js [hunk 7]
```diff
@@ -187,6 +187,7 @@
       var transferSizeMbs = response.loaded * 8 / 1000000;
       var transferDurationSeconds = this.totalTime/1000;
       result.bandwidth = transferSizeMbs / transferDurationSeconds;
+      result.loaded = response.loaded;
```
Comment (kemarc) [inline public/lib/xmlhttprequest.js]: please add comment explaining this line.

File: public/lib/xmlhttprequest.js [hunk 8]
```diff
@@ -208,6 +209,10 @@
           var transferSizeMbs = ((response.loaded - this.prevLoad) * 8) / 1000000;
           var transferDurationSeconds = result.totalTime/1000;
           result.bandwidth = transferSizeMbs/transferDurationSeconds;
+            result.loaded = response.loaded;
```
Comment (kemarc) [inline public/lib/xmlhttprequest.js]: is there a reason this is being set to itself?

File: public/speed-testJS.js [hunk 9]
```diff
@@ -379,91 +396,114 @@
...
+            prevDownloadSize = result.prevDownloadSize;
+            downloadTestTimeout = result.timeout;
+            if (result.calculateResults) {
+                //var results = testResults.join("\",\"");
+                //console.log(results);
+                var calculateMeanStats = new window.calculateStats('http://' + testPlan.baseUrlIPv4 + '/calculator', testResults, calculateStatsonComplete, calculateStatsonError);
+                calculateMeanStats.performCalculations();
+            } else {
+                //downloadTest(version);
+                downloadTest(version === 'IPv6' ? 'IPv6' : 'IPv4');
```
Comment (kemarc) [inline public/speed-testJS.js]: storing this in a semantically named variable maybe easier to read

File: public/examples/download/downloadApp.js (hunk 4 above)
Comment (Pradhyumna02) [inline public/examples/download/downloadApp.js]: yes.. when we click start test again..  we need to reset them

File: public/lib/xmlhttprequest.js (hunk 8 above)
Comment (Pradhyumna02) [inline public/lib/xmlhttprequest.js]: it's not set to itself

result and response

**Comment by kemarc** [comment]
+1

**Comment by kemarc** [comment]
still working on this?

**Pull request closed without merging**